### ISSUE TITLE
Get SOTA model results for comparison

### ISSUE THREAD

**Issue by nclgbd** [nclgbd/ResearchToolKit#3]
# Overview
It seems that the SOTA models available are credited as being [Lion](https://paperswithcode.com/paper/symbolic-discovery-of-optimization-algorithms#code) in conjuction with [CoCa](https://paperswithcode.com/paper/coca-contrastive-captioners-are-image-text)

## Model candidates
* ViT-B/16_384
* CoAtNet-1
* CoCa

**Comment by nclgbd** [comment]
# Notes from Lion (Chen et al.)

![Image](https://github.com/nclgbd/ResearchToolKit/assets/38570878/bf8a4110-d077-46bb-a5c8-c90b16462fb1)


![Image](https://github.com/nclgbd/ResearchToolKit/assets/38570878/fc040acd-4a67-4bcf-a035-2d2e89bf25d1)

**Comment by nclgbd** [comment]
# Notes from CoCa

**Comment by nclgbd** [comment]
A note on CLIP and ALIGN:
![image](https://github.com/nclgbd/ResearchToolKit/assets/38570878/b8c708f3-df0e-48e0-a5ed-42ea5035075a)

**Comment by nclgbd** [comment]
# [Run summary](https://ml.azure.com/experiments/id/1c20df93-62d0-4288-bfce-76336d94b0bf/runs/e556645a-6fc2-4a98-867b-66cff9fba2cd?wsid=/subscriptions/a3568436-9c4c-4f30-a6ce-03452353152c/resourcegroups/nguob-rg/providers/Microsoft.MachineLearningServices/workspaces/master-thesis&tid=f6b6dd5b-f02f-441a-99a0-162ac5060bd2)

## Test results
### Evaluation metrics
| model_name                |   roc_auc |   accuracy |      loss |   precision |   recall |   f1-score |
|:--------------------------|----------:|-----------:|----------:|------------:|---------:|-----------:|
| vit_base_patch16_clip_384 |   0.97265 |   0.974359 | 0.0743239 |     0.97265 |  0.97265 |    0.97265 |

### Confusion matrix
|           |   NORMAL |   PNEUMONIA |
|:----------|---------:|------------:|
| NORMAL    |      226 |           8 |
| PNEUMONIA |        8 |         382 |

## `config.yaml`
```yaml
# --config-name=config

datasets:
  dataloader:
    _target_: monai.data.ThreadDataLoader
    batch_size: 16
    num_workers: 24
  dim: 384
  encoding:
    0: NORMAL
    1: PNEUMONIA
  extension: .jpeg
  instantiate:
    _target_: monai.data.ImageDataset
  labels:
  - NORMAL
  - PNEUMONIA
  patient_data: null
  preprocessing:
    resample_value: 1
    sample_to_value: 3500
    use_sampling: true
  scan_data: /home/nicoleg/workspaces/teddytoolkit/.data/Chest_XRay_Images_RGB/
  transforms:
    load:
    - _target_: monai.transforms.ScaleIntensity
    - _target_: monai.transforms.EnsureChannelFirst
    - _target_: monai.transforms.EnsureType
      track_meta: true
    - _target_: monai.transforms.Resize
      spatial_size:
      - ${datasets.dim}
      - ${datasets.dim}
    train:
    - _target_: torchvision.transforms.RandomResizedCrop
      antialias: true
      scale:
      - 0.75
      - 1.25
      size:
      - ${datasets.dim}
      - ${datasets.dim}
    - _target_: torchvision.transforms.RandomAffine
      degrees: 45
      shear: 16
      translate:
      - 0.15
      - 0.2
date: ${now:%Y-%m-%d}
ignite:
  checkpoint:
    _target_: ignite.handlers.checkpoint.Checkpoint
    filename_prefix: best
    n_saved: 10
    save_handler: artifacts/checkpoints/
  early_stopping:
    _target_: ignite.handlers.early_stopping.EarlyStopping
    patience: 5
  lr_scheduler: null
  metrics:
    Accuracy: null
    Loss: null
    Predictions: null
  score_name: loss
  use_checkpoint: true
  use_early_stopping: false
  use_lr_scheduler: false
  use_multi_gpu: false
job:
  device: 1
  dry_run: false
  epoch_length: null
  max_epochs: 25
  perform_validation: true
  prepare_function:
    _target_: rtk.ignite.prepare_run
  random_state: 101
  set_track_meta: false
  train_test_split:
    shuffle: true
    test_size: 0.1
  use_autocast: true
  use_azureml: true
  use_mlflow: true
  use_pretrained: true
  use_transforms: true
mlflow:
  tracking_uri: file:///home/nicoleg/workspaces/teddytoolkit/mlruns/
models:
  criterion:
    _target_: torch.nn.CrossEntropyLoss
  model:
    _target_: timm.create_model
    model_name: vit_base_patch16_clip_384
    num_classes: 2
    pretrained: ${job.use_pretrained}
  optimizer:
    _target_: torch.optim.Adam
    lr: 1.0e-05
    weight_decay: 0.0
postfix: ''
sklearn:
  model_selection:
    train_test_split:
      shuffle: true
      test_size: 0.1
timestamp: ${now:%H-%M-%S}

```

**Comment by nclgbd** [comment]
# [Run summary](https://ml.azure.com/experiments/id/1c20df93-62d0-4288-bfce-76336d94b0bf/runs/25b48e77-ceb0-463c-87a3-fa3c02150b66?wsid=/subscriptions/a3568436-9c4c-4f30-a6ce-03452353152c/resourceGroups/nguob-rg/providers/Microsoft.MachineLearningServices/workspaces/master-thesis&tid=f6b6dd5b-f02f-441a-99a0-162ac5060bd2#overview) for Chest X-Ray14 dataset

## Test results
### Evaluation metrics
| model_name                |   roc_auc |   accuracy |     loss |   precision |   recall |   f1-score |
|:--------------------------|----------:|-----------:|---------:|------------:|---------:|-----------:|
| vit_base_patch16_clip_384 |  0.816779 |   0.966314 | 0.162491 |     0.86287 | 0.816779 |   0.838009 |

### Confusion matrix
|            |   No Finding |   Pneumonia |
|:-----------|-------------:|------------:|
| No Finding |         9726 |         135 |
| Pneumonia  |          218 |         400 |

## `config.yaml`
```yaml
# --config-name=config

datasets:
  additional_datasets:
    dataset_configs:
    - filepath: /home/nicoleg/workspaces/teddytoolkit/configs/datasets/chest-xray-dataset.yaml
      loader: load_chest_xray_dataset
  dataloader:
    _target_: monai.data.ThreadDataLoader
    batch_size: 16
    num_workers: 24
  dim: 384
  encoding:
    No Finding: 0
    Pneumonia: 1
  extension: .png
  index: Image Index
  instantiate:
    _target_: monai.data.ImageDataset
  labels:
  - No Finding
  - Pneumonia
  patient_data: /home/nicoleg/workspaces/teddytoolkit/.data/Chest_XRay_14_Kaggle_RGB/Data_Entry_2017.csv
  preprocessing:
    sampling_method:
      method:
        _partial_: true
        _target_: imblearn.datasets.make_imbalance
        verbose: true
      sample_to_value: 5000
    subset:
    - Atelectasis
    - Edema
    - Effusion
    - Consolidation
    - Pneumonia
    use_sampling: true
  scan_data: /home/nicoleg/workspaces/teddytoolkit/.data/Chest_XRay_14_Kaggle_RGB/
  target: Finding Labels
  transforms:
    load:
    - _target_: monai.transforms.ScaleIntensity
    - _target_: monai.transforms.EnsureChannelFirst
    - _target_: monai.transforms.EnsureType
      track_meta: true
    - _target_: monai.transforms.Resize
      spatial_size:
      - ${datasets.dim}
      - ${datasets.dim}
    train:
    - _target_: torchvision.transforms.RandomResizedCrop
      antialias: true
      scale:
      - 0.75
      - 1.25
      size:
      - ${datasets.dim}
      - ${datasets.dim}
    - _target_: torchvision.transforms.RandomAffine
      degrees: 45
      shear: 16
      translate:
      - 0.15
      - 0.2
date: ${now:%Y-%m-%d}
ignite:
  checkpoint:
    _target_: ignite.handlers.checkpoint.Checkpoint
    filename_prefix: best
    n_saved: 10
    save_handler: artifacts/checkpoints/
  early_stopping:
    _target_: ignite.handlers.early_stopping.EarlyStopping
    patience: 5
  log_interval: 5
  lr_scheduler: null
  metrics:
    Accuracy: null
    Loss: null
    Predictions: null
  score_name: loss
  use_checkpoint: true
  use_early_stopping: true
  use_lr_scheduler: false
  use_multi_gpu: false
job:
  device: 0
  dry_run: false
  epoch_length: null
  max_epochs: 100
  perform_validation: true
  prepare_function:
    _target_: rtk.ignite.prepare_run
  random_state: 101
  set_track_meta: false
  train_test_split:
    shuffle: true
    test_size: 0.1
  use_autocast: true
  use_azureml: true
  use_mlflow: true
  use_pretrained: true
  use_transforms: true
mlflow:
  tracking_uri: file:///home/nicoleg/workspaces/teddytoolkit/mlruns/
models:
  criterion:
    _target_: torch.nn.CrossEntropyLoss
  model:
    _target_: timm.create_model
    model_name: vit_base_patch16_clip_384
    num_classes: 2
    pretrained: ${job.use_pretrained}
  optimizer:
    _target_: torch.optim.Adam
    lr: 1.0e-05
    weight_decay: 0.001
postfix: ''
sklearn:
  model_selection:
    train_test_split:
      shuffle: true
      test_size: 0.1
timestamp: ${now:%H-%M-%S}

```
